geode-gfsh/src/main/java/org/apache/geode/management/internal/cli/functions/DataCommandFunction.java

```diff
@@ -237,6 +238,9 @@ private DataCommandResult select(InternalCache cache, Object principal, String q
       if (queryObserver != null) {
         QueryObserverHolder.reset();
       }
+      if (tracedQuery.isTraced()) {
```

Comment:
this block is already inside the `if (tracedQuery.isTraced())` condition, is this `if` necessary?

Reply:
It is not. This finally block is outside any `if (tracedQuery.isTraced())` condition.